src/main/java/wolox/training/services/impl/BookServiceImpl.java

```diff
@@ -68,9 +71,10 @@ public Book createByOpenLibrary(String isbn) {
     }
 
     @Override
-    public List<Book> findByAllFields(String genre, String author, String image, String title,
-        String subtitle, String publisher, String year, int pages, String isbn) {
-        return bookRepository.findByAllFields(genre, author, image, title, subtitle, publisher, year, pages, isbn).orElseThrow(BookNotFoundException::new);
+    public Page<Book> findByAllFields(String genre, String author, String image, String title,
+        String subtitle, String publisher, String year, int pages, String isbn, int page, int size, String sortBy, String order) {
+        PageRequest pageRequest = PageRequest.of(page, size, order.equalsIgnoreCase("asc") ? Sort.by(sortBy).ascending() : Sort.by(sortBy).descending());
```

Comment:
the order logic could be in a private method

Reply:
Order logic separated to private method